apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/store/db/DbValidatorInternalStore.scala

```diff
@@ -4,73 +4,139 @@
 package org.lfdecentralizedtrust.splice.validator.store.db
 
 import cats.data.OptionT
+import com.digitalasset.canton.lifecycle.{CloseContext, FutureUnlessShutdown}
+import com.digitalasset.canton.logging.{ErrorLoggingContext, NamedLoggerFactory, NamedLogging}
 import com.digitalasset.canton.discard.Implicits.DiscardOps
-import com.digitalasset.canton.lifecycle.CloseContext
-import com.digitalasset.canton.logging.{NamedLoggerFactory, NamedLogging}
 import com.digitalasset.canton.resource.DbStorage
 import com.digitalasset.canton.tracing.TraceContext
+import org.lfdecentralizedtrust.splice.util.FutureUnlessShutdownUtil.futureUnlessShutdownToFuture
+import com.digitalasset.canton.topology.ParticipantId
+import org.lfdecentralizedtrust.splice.validator.store.{ValidatorInternalStore, ValidatorStore}
 import io.circe.syntax.EncoderOps
 import io.circe.{Decoder, Encoder, Json}
 import org.lfdecentralizedtrust.splice.store.db.AcsJdbcTypes
-import org.lfdecentralizedtrust.splice.util.FutureUnlessShutdownUtil.{
-  futureUnlessShutdownToFuture,
-  optionTfutureUnlessShutdownToFuture,
-}
-import org.lfdecentralizedtrust.splice.validator.store.ValidatorInternalStore
 import slick.jdbc.JdbcProfile
 import slick.jdbc.canton.ActionBasedSQLInterpolation.Implicits.actionBasedSQLInterpolationCanton
-
 import scala.concurrent.{ExecutionContext, Future}
+import org.lfdecentralizedtrust.splice.store.db.{StoreDescriptor, StoreDescriptorStore}
+import java.util.concurrent.atomic.AtomicReference
 
 class DbValidatorInternalStore(
+    participant: ParticipantId,
+    key: ValidatorStore.Key,
     storage: DbStorage,
-    protected val loggerFactory: NamedLoggerFactory,
-)(implicit ec: ExecutionContext, closeContext: CloseContext)
-    extends ValidatorInternalStore
+    val loggerFactory: NamedLoggerFactory,
+)(implicit
+    val ec: ExecutionContext,
+    val loggingContext: ErrorLoggingContext,
+    val closeContext: CloseContext,
+) extends ValidatorInternalStore
     with AcsJdbcTypes
     with NamedLogging {
 
   val profile: JdbcProfile = storage.profile.jdbc
+  private val storeDescriptor = StoreDescriptor(
+    version = 2,
+    name = "DbValidatorInternalConfigStore",
+    party = key.validatorParty,
+    participant = participant,
```

Comment:
let's please remove the participant id, I don't see the need for duplication with the validator party